File: ProjectCS/UI/LoopPage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SessionResultsCS.UI
{
    public abstract class LoopPage: IPage
    {
        private bool _looping;

        private MenuActionPicker? _picker;

        public void Start()
        {
            _looping = true;
            while (_looping)
            {
                Loop();
                Console.WriteLine("Действия: ");
                _picker?.PickAndRun();
            }
        }

        protected abstract void Loop();

        protected void SetPicker(MenuActionPicker picker)
        {
            _picker = picker;
        }

        protected void StopLoop()
        {
            _looping = false;
        }
    }
}

File: Project/UI/LoopPage.cpp
#include "UI/LoopPage.hpp"

#include <iostream>


void LoopPage::Start() {
	m_looping = true;
	while (m_looping) {
		Loop();
		std::cout << "Действия:\n";
		if (m_picker != nullptr) {
			m_picker->PickAndRun();
		}
	}
}

void LoopPage::StopLoop() {
	m_looping = false;
}

void LoopPage::SetPicker(std::unique_ptr<MenuActionPicker> picker) {
	m_picker = std::move(picker);
}
